internal/operators/ocs/ocs_operator.go

```diff
@@ -105,26 +105,25 @@ func (o *operator) GetHostRequirements(_ context.Context, cluster *common.Cluste
 
 	disks := int64(getValidDiskCount(inventory.Disks) - 1)
 
-	if numOfHosts < 3 {
-		return nil, fmt.Errorf("OCS requires a minimum of 3 hosts")
-	}
-
-	if numOfHosts == 3 { // Compact Mode
+	if numOfHosts <= 3 { // Compact Mode
+		var reqDisks int64 = 1
+		if disks > 0 {
+			reqDisks = disks
+		}
+		// for each disk ocs requires 2 CPUs and 5 GB RAM
 		if host.Role == models.HostRoleMaster || host.Role == models.HostRoleAutoAssign {
-			if disks <= 0 {
-				return nil, fmt.Errorf("OCS requires a minimum of one non-bootable disk per host")
-			}
 			return &models.ClusterHostRequirementsDetails{
-				CPUCores:   CPUCompactMode + disks*o.config.OCSRequiredDiskCPUCount,
-				RAMMib:     conversions.GbToMib(MemoryGBCompactMode + disks*o.config.OCSRequiredDiskRAMGB),
+				CPUCores:   CPUCompactMode + reqDisks*o.config.OCSRequiredDiskCPUCount,
+				RAMMib:     conversions.GbToMib(MemoryGBCompactMode + reqDisks*o.config.OCSRequiredDiskRAMGB),
 				DiskSizeGb: MinDiskSize,
 			}, nil
 		}
-		return nil, fmt.Errorf("OCS compact mode unsupported role: %s", host.Role)
-	}
-
-	if host.Role == models.HostRoleAutoAssign { //If auto-assign we cannot proceed with minimal or standard mode
-		return nil, fmt.Errorf("OCS minimal mode unsupported role: %s", host.Role)
+		// regular worker req
+		return &models.ClusterHostRequirementsDetails{
+			CPUCores:   CPUMinimalMode + reqDisks*o.config.OCSRequiredDiskCPUCount,
+			RAMMib:     conversions.GbToMib(MemoryGBMinimalMode + reqDisks*o.config.OCSRequiredDiskRAMGB),
+			DiskSizeGb: MinDiskSize,
+		}, nil
```

Comment:
@jakub-dzon  for len(hosts) <=3 , will there be a case with role worker? I think we should return Compact mode requirements for `len(hosts) <=3 ` ,  irrespective of role. WDYT?

Reply:
If user chose worker role deliberately than I think the requirements should be for worker; user might add more hosts later and change use case to standard.

Reply:
Ack.